### ISSUE TITLE
Multiple-thread parallelization not helping

### ISSUE THREAD

**Issue by AABoyles** [CDCgov/MicrobeTrace#51]
When I compute 35000 links in MicrobeTrace using 1 core, it takes around 30 seconds. When I perform the same analysis using 30 cores, I would expect to see it returning on the order of 1 sec. (but I wouldn't worry if it was as high as 5-6 seconds, lost in transit and reassembly times). In reality it takes... 30 seconds. Our parallelization scheme seems to be offering no benefit beyond the first thread. I don't know why this is the case, but I'm guessing that a library like [Hamsters.io](https://www.hamsters.io/) has already solved this problem. Let's look at transitioning to one of those instead of grinding away for no benefit.

**Comment by AABoyles** [comment]
Belay that, it looks like [hamsters suffers from the same problem](https://www.hamsters.io/examples/squareroot). Maybe this is a Chrome thing?

**Comment by AABoyles** [comment]
Nope, looks like Firefox does it too. :(

**Comment by AABoyles** [comment]
OK, gonna keep an eye on this one for awhile, but not change anything for now.

**Comment by AABoyles** [comment]
After a bit of detective work, it appears that our bottlenecks are not so much computing links as transferring them and matching them to existing links in the dataset. We've optimized the computations, and managed the transfer porblem by leveraging [transferrables](https://developer.mozilla.org/en-US/docs/Web/API/Transferable). The link matching problem we've cheated away by checking to see if it's possible that a link already exists and only matching if it is and just pushing it if it's not (This may be a problem in cases for which multiple sequences share a common ID, but I'm not going to worray about this edge-case for now). Together, these yield an immense speedup.

The next steps have already been outlined in #52 and #56.
